cotel/src/main/scala/day2/SpreadsheetChecksum.scala

```diff
@@ -0,0 +1,35 @@
+package day2
+
+
+object SpreadsheetChecksum {
+  type Row = List[Int]
+  type SpreadSheet = List[Row]
+
+  def Row(xs: Int*) = List(xs: _*)
+
+  def SpreadSheet(rows: Row*) = List(rows: _*)
+
+  def checksum(spreadSheet: SpreadSheet): Int = {
+    spreadSheet
+      .map { list => (list.max, list.min) }
+      .foldLeft(0) {
+        case (acc, (max, min)) => acc + (max - min)
+      }
+  }
+
+  def divsum(spreadSheet: SpreadSheet): Int = {
+    spreadSheet
+      .map(findEvenlyDivision)
+      .foldLeft(0) {
+        case (acc, Some(x)) => acc + x.toInt
+        case (acc, None) => acc
+      }
+  }
+
+  private def findEvenlyDivision(row: Row): Option[Int] = {
```

Comment:
6 iterations in a single method

Reply:
Yep, I'm not so happy with today's solution for this part. I think the problem was I don't really know how to use `for comprehensions` in Scala but I'm pretty sure I can solve it with it instead of using collection operators.